torch/distributed/CONTRIBUTING.md

```diff
@@ -6,72 +6,63 @@ Please go through PyTorch's top level [Contributing Guide](../../CONTRIBUTING.md
+- Mixed Precision Hooks: [torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py](https://github.com/pytorch/pytorch/blob/main/torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py)
+#### Fully Sharded Data Parallel (FSDP)
 
-[request_callback.h](../csrc/distributed/rpc/request_callback.h) and [request_callback_impl.h](../csrc/distributed/rpc/request_callback_impl.h) deal with how to handle RPC calls on remote servers.
+- FSDP: [torch/distributed/fsdp/api.py](https://github.com/pytorch/pytorch/blob/main/torch/distributed/fsdp/api.py)
+- FSDP2: [torch/distributed/_composable/fsdp/fully_shard.py](https://github.com/pytorch/pytorch/blob/main/torch/distributed/_composable/fsdp/fully_shard.py)
+- Implementations are contained in other files in the same folder as the API for each variant
 
-### Remote Reference (RRef)
+#### Tensor Parallel (TP)
 
-Most of the APIs for RRefs can be found in [rpc/api.py](rpc/api.py). The C++ interface can be found in [rref_interface.h](../../aten/src/ATen/core/rref_interface.h) and implementations in [rref_impl.h](../csrc/distributed/rpc/rref_impl.h) and [rref_context.h](../csrc/distributed/rpc/rref_context.h).
+- API layer: [torch/distributed/tensor/parallel/api.py](https://github.com/pytorch/pytorch/blob/main/torch/distributed/tensor/parallel/api.py)
+- Implementation: see other files in the same folder
 
-### Distributed Autograd
+#### Pipeline Parallel (PP)
 
-The top level APIs for distributed autograd can be found in [distributed/autograd/init.py](autograd/__init__.py) and [distributed/autograd/init.cpp](../csrc/distributed/autograd/init.cpp).
+- Pipeline Schedules: [torch/distributed/pipelining/schedules.py](https://github.com/pytorch/pytorch/blob/main/torch/distributed/pipelining/schedules.py)
+- Pipeline Stage: [torch/distributed/pipelining/stage.py](https://github.com/pytorch/pytorch/blob/main/torch/distributed/pipelining/stage.py)
 
-The core engine for executing a distributed backward pass can be found in [dist_engine.h](../csrc/distributed/autograd/engine/dist_engine.h)
 
-### Distributed Optimizer
+## Adding Tests
 
-The distributed optimizer is completely written in Python and can be found at [optimizer.py](optim/optimizer.py)
+You should write tests for your changes just like in other parts of PyTorch, but you may need to use some test infrastructure to run either multi-process tests on multiple GPUs, or use a FakeProcessGroup to mock out communications.
 
-### Onboarding Tasks
+Most testing can be done from python, and you can find existing python tests [here](https://github.com/pytorch/pytorch/tree/main/test/distributed).
 
-A list of onboarding tasks can be found [here](https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3Apt_distributed_rampup+).
+For an example of using the MultiProcessTestCase to run a test on multiple GPUs, see tests in [test_c10d_nccl.py](https://github.com/pytorch/pytorch/blob/main/test/distributed/test_c10d_nccl.py)
 
```

Comment:
Shall we also add cpp related test and how to run them? 

For example, for test `test/cpp/c10d/ProcessGroupNCCLErrorsTest.cpp`

We can either run `./build/bin/ProcessGroupNCCLTest --gtest_filter=ProcessGroupNCCLErrorsTest.testBroadcast`

or `python test/run_test.py --cpp --verbose -i cpp/ProcessGroupNCCLErrorsTest`

The former is still valuable because we can just run one test rather than the whole test.

Reply:
well, i ran out of time on this and just decided to merge it. i'd be happy to stamp another PR to add more info to the tests.